src/main/java/org/broadinstitute/consent/http/service/DatasetService.java

```diff
@@ -435,6 +437,23 @@ public List<Map<String, String>> autoCompleteDatasets(String partial, Integer da
         ).collect(Collectors.toList());
     }
 
+    public Dataset approveDataset(Dataset dataset, User user, Boolean approvalBool) {
+        Boolean currentApprovalState = dataset.getDacApproval();
+        Integer datasetId = dataset.getDataSetId();
+        Dataset datasetReturn = dataset;
+        //Only update and fetch the dataset if it hasn't already been approved
+        //If it has, simply returned the dataset in the argument (which was already queried for in the resource)
+        if(Objects.isNull(currentApprovalState) || !currentApprovalState) {
+            datasetDAO.updateDatasetApproval(approvalBool, Instant.now(), user.getUserId(), datasetId);
+            datasetReturn = datasetDAO.findDatasetById(datasetId);
+        } else {
+            if(Objects.isNull(approvalBool) || !approvalBool) {
+                throw new ForbiddenException("Dataset is already approved");
+            }
```

Comment:
I think we could use a different exception here. `ForbiddenException` implies the user has no permissions to perform this function, but they do, this is just a disallowed case. I think `IllegalArgumentException` might be a better fit since those end up being converted to `BadRequestException` at the resource level.